accepted/0019-app-notifications.md

```diff
@@ -0,0 +1,111 @@
+
+For developers, making sure this is well documented is important. Knowing that they can reach out to the user within the UI when things go wrong is important.
+
+# Drawbacks
+
+Complexity. This will require a lot of code, logic, and a background runner keeping track of active/dismissed alerts. The solution to this may be the option to turn it off somehow for bigger productions that only need issues confirmed before going live.
+
+Visual noise. This one is less of a concern, but at the same time we don't want too much going on in the UI. A big toolbar along the top could be distracting, but at the same time going subtle means that the user could potentially not realise that something has gone wrong.
+
+# Additional Information
+
+Alerts shouldn't have a timeout, so that they're not missed by the user.
+
+There are many things to consider in this design. It's highly likely that the backend & UI would be developed & PR'd independently rather than together. This'd ensure a solid backend design, and an easier time reviewing it.
+
+The backend design & scope should be fleshed out significantly more before a real impelementation is started.
+
+# Alternatives
+
+- Exposing QSystemTrayIcon's `showMessage` - providing plugins a way to use native notifications.
+  - Downsides include:
+    - They will trigger a sound that the user cannot mute
+    - Windows automatically disables these when a game is running (via Focus assist)
+    - they are distracting and interrupt the user
+    - Due to how they're visually exposed to the user, they could be abused by plugins ("annoying")
+  - Upsides include:
+    - easier to implement
+    - will get the user's attention
+
+# Unresolved questions
+
+* Should Alerts have a list of methods to dismiss (scene collection switch, profile switch)?
+  * Methods to dismiss would be quite limited in libobs, and would add unneeded complexity to the frontend. Methods to dismiss would instead be tied to sources, outputs, etc.
+* The focus of this current design is of the assumption that alerts are only shown in the main window. Often it makes more sense to show warnings/info within Properties themselves. I personally think the scope should be limited to the core alert system, and overhauling Properties to support many more things is a task for another day. Thoughts?
+* Rather than making the summary/description link/anchor capable, should we instead provide a "link" field, and auto generate a "Learn more" link at the end of the summary? The big downside of this is then only one link can be provifded, rather than providing extensibility and further information for the user.
+* Should this Alert API contact an OBS server to recieve alerts? Use case here would be if a service is having connectivity issues that is out of our control. This could explicitly be optional, maybe only enabled when the user links their account with a service.
```

Comment:
Probably also out of scope for the RFC, but it would be cool to see service plugins add support for the alert system after the fact.